src/utils/wishlistUtils.js

```diff
@@ -0,0 +1,57 @@
+import { actionTypes } from "../context/dataContext/actionTypes";
+import { addToWishlist,removeFromWishlist } from "../network";
+
+const addToWishlistHandler = async (dispatch, product, token) => {
+  dispatch({ type: actionTypes.fetchData });
+  try {
+    const { data } = await addToWishlist(product, token);
+    dispatch({ type: actionTypes.setWishlistData, payload: data.wishlist });
+    dispatch({ type: actionTypes.fetchSuccess });
+  } catch (error) {
+    dispatch({ type: actionTypes.fetchFailed });
+  }
+};
+const addToWishlistClickHandler = (
+  selectedItem,
+  dispatch,
+  token,
+  productData
+) => {
+  const { _id, productName, price, image } = selectedItem;
+  const addItemObj = {
+    _id,
+    productName,
+    price,
+    image,
+  };
+  addToWishlistHandler(dispatch, addItemObj, token);
+  let newData = productData.map((item) => {
+    if (item._id === selectedItem._id) {
+      item.addedToWishlist = true;
+    }
+    return item;
+  });
+  dispatch({ type: actionTypes.setProductData, payload: newData });
+};
+const removeFromWishlistHandler = async (dispatch, productId, token) => {
+  dispatch({ type: actionTypes.fetchData });
+  try {
+    const { data } = await removeFromWishlist(productId, token);
+    dispatch({ type: actionTypes.setWishlistData, payload: data.wishlist });
+    dispatch({ type: actionTypes.fetchSuccess });
+  } catch (error) {
+    dispatch({ type: actionTypes.fetchFailed });
+  }
+};
+
+const removeFromWishlistClickHandler = (id, token, productData, dispatch) => {
+  removeFromWishlistHandler(dispatch, id, token);
+  let newData = productData.map((item) => {
+    if (item._id === id) {
```

Comment:
this can be written as 
const newData = productData.map((item) => item._id === id &&  item.addedToWishlist = false );

Reply:
I am not sure if we can change the property of an object like this because I want to set the property of item object to false here if the condition is met. But will try to see if I can do this.